uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/manage-leave-requests.component.js

```diff
@@ -254,36 +254,43 @@ define([
     }
 
     /**
-     * Loads the managees and calls loadLeaveRequests()
+     * Loads the leave requests for the given data type
      *
+     * @param {String} dataType - which data to load, table or filter or both
      * @return {Promise}
      */
-    function loadManageesAndLeaves () {
-      vm.loading.content = true;
+    function loadData (dataType) {
+      if (dataType) {
+        return loadLeaveRequests(dataType);
+      } else {
+        return $q.all([
+          loadLeaveRequests('table'),
+          loadLeaveRequests('filter')
+        ]);
+      }
+    }
 
+    /**
+     * Loads the managees and calls loadLeaveRequests()
+     *
+     * @param {String} dataType - which data to load, table or filter
+     * @return {Promise}
+     */
+    function loadManageesAndLeaves (dataType) {
```

Comment:
Why do you need to bundle them together? Do you *always* need to fetch first the contacts and then the leave requests?